File: src/com/leetcode/Easy/BestTimeToBuyAndSellStock.java
package com.leetcode.Easy;

// 121. Best Time to Buy and Sell Stock
public class BestTimeToBuyAndSellStock {
    public int maxProfit(int[] prices) {
        if (prices.length == 0) return 0;

        int maxProfit = 0, minStock = prices[0];
        for (int price : prices) {
            if (price > minStock) {
                maxProfit = Math.max(maxProfit, price - minStock);
            } else {
                minStock = price;
            }
        }
        return maxProfit;
    }
}

File: JavaScript/Easy/BestTimeToBuyAndSellStock.js
/**
 * 121. Best Time to Buy and Sell Stock
 * @param {number[]} prices
 * @return {number}
 */
var maxProfit = function (prices) {
  if (prices.length == 0) return 0;

  let minPrice = prices[0];
  let maxProfit = 0;
  for (let i = 0; i < prices.length; i++) {
    let price = prices[i];
    if (price > minPrice) {
      maxProfit = Math.max(maxProfit, price - minPrice);
    } else {
      minPrice = price;
    }
  }
  return maxProfit;
};
